src/api/http.rs

```diff
@@ -102,19 +117,83 @@ fn vdisks(bob: State<BobSrv>) -> Json<Vec<VDisk>> {
 }
 
 #[get("/vdisks/<vdisk_id>")]
-fn vdisk_by_id(bob: State<BobSrv>, vdisk_id: u32) -> Json<Option<VDisk>> {
-    let vdisk = get_vdisk_by_id(&bob, vdisk_id);
-    Json(vdisk)
+fn vdisk_by_id(bob: State<BobSrv>, vdisk_id: u32) -> Option<Json<VDisk>> {
+    get_vdisk_by_id(&bob, vdisk_id).map(Json)
 }
 
 #[get("/vdisks/<vdisk_id>/partitions")]
-fn partitions(_bob: State<BobSrv>, vdisk_id: usize) -> String {
-    format!("partitions of vdisk {}", vdisk_id)
+fn partitions(bob: State<BobSrv>, vdisk_id: u32) -> Result<Json<VDiskPartitions>, Status> {
+    let backend = &bob.grinder.backend.backend();
+    debug!("get backend: OK");
+    let groups = backend.vdisks_groups().ok_or_else(|| {
+        warn!("only pearl backend supports partitions");
+        Status::NotFound
```

Comment:
- Here should be NotAcceptable (406) status code.
 - Also add the same text as you print to log - to statuses: Status::NotAcceptable("Only pearl backend supports partitions")